### ISSUE TITLE
Deno support

### ISSUE THREAD

**Issue by ProvocaTeach** [belayeng/quadstore#139]
Hello, I am opening an issue for supporting [Deno](https://deno.land) via ES module importing out-of-the-box, per discussion in issue #138 at the suggestion of @jacoscaz.

Deno is a runtime for JavaScript created by Ryan Dahl (who also built Node). Benefits of Deno include:

- Direct support for TypeScript
- Security by default
- Browser-like behavior whenever possible

Members of the Quadstore team may have a better idea than I do of how far-off this is. I don’t have a great sense of how much the most popular RDF/JS projects depend on Node. However, [Skypack](https://skypack.dev) or a similar Deno-friendly CDN may be a good way to provide stopgap support.

Some helpful Deno links:

- [Deno homepage](https://deno.land)
- [Deno v1.13.2 manual](https://deno.land/manual@v1.13.2)
  - [Section on module system](https://deno.land/manual@v1.13.2/linking_to_external_code)
  - [Section on using npm / Node.js code](https://deno.land/manual@v1.13.2/npm_nodejs)
  - [Section on Deno-friendly CDNs, e.g. Skypack](https://deno.land/manual@v1.13.2/npm_nodejs/cdns)
- [Skypack docs](https://docs.skypack.dev)
  - [Package checks](https://docs.skypack.dev/package-authors/package-checks): Best practices for Node package authors wishing to support ES modules.
  - [Deno support](https://docs.skypack.dev/skypack-cdn/code/deno): How Skypack can be used for Deno specifically, and when this is/isn’t possible.
- [10 Things I Regret About Node.js - Ryan Dahl - JSConf EU](https://www.youtube.com/watch?v=M3BM9TB-8yA): Helpful for understanding the philosophy and design goals of Deno.
- [Syntax.fm episode on Deno](https://syntax.fm/show/322/the-deno-show): A somewhat-silly but comprehensive podcast introduction to Deno that assumes no prior knowledge.

**Comment by jacoscaz** [comment]
Thanks to the recent addition of separate ESM vs. CJS builds, partial support for Deno is now available through Skypack in `quadstore@11.0.0-beta.5`: 

```ts
import { DataFactory } from 'https://cdn.skypack.dev/rdf-data-factory@1.1.1';
import { Quadstore } from 'https://cdn.skypack.dev/quadstore@11.0.0-beta.4';
import { MemoryLevel } from 'https://cdn.skypack.dev/memory-level@1.0.0';

const backend = new MemoryLevel();
const dataFactory = new DataFactory();
const store = new Quadstore({ backend, dataFactory });

await store.open();
await store.put(dataFactory.quad(
  dataFactory.namedNode('ex://s'),
  dataFactory.namedNode('ex://p'),
  dataFactory.namedNode('ex://o'),
));
const { items } = await store.get({});
console.log(items);
await store.close();
```

Further work is needed to add support for Deno in `quadstore-comunica`, thus enabling SPARQL queries in Deno, and likely depends on fixing this upstream issue: https://github.com/RubenVerborgh/SPARQL.js/issues/139 .

**Comment by jacoscaz** [comment]
Closing - will be a part of the upcoming `11.0.0` release!
